### PR TITLE
release: 24.11.1

### PR DESCRIPTION
**Pull request by adwk67**
> [!CAUTION]
> ## DO NOT MERGE MANUALLY!
> This branch will be merged (and the commit tagged) by stackable-utils once any necessary commits have been cherry-picked to here from the main branch.

### CODE REVIEW

**Comment by adwk67** [comment]
### Integration tests
Local release.yaml edited as below to use images for the 24.11.1 PRs:

```
# Contains all operators required to run the test suite.
---
releases:
  # Do not change the name of the release as it's referenced from run_tests.sh
  tests:
    releaseDate: 1970-01-01
    description: Integration test
    products:
      commons:
        operatorVersion: 24.11.1-pr317
      secret:
        operatorVersion: 24.11.1-pr540
      listener:
        operatorVersion: 24.11.1-pr266
      airflow:
        operatorVersion: 24.11.1-pr567
      spark-k8s:
        # corner-case not 24.11.1-pr513 as the DAG calls spark implicitly with 
        # product image selection (i.e. bypasses test-definition.yaml)
        operatorVersion: 24.11.0
```

Results:


PRs
  - airflow-operator 567
  - commons-operator 317
  - druid-operator 675
  - hbase-operator 610
  - hdfs-operator 638
  - hello-world-operator 150
  - hive-operator 558
  - kafka-operator 808
  - listener-operator 266
  - nifi-operator 735
  - opa-operator 674
  - secret-operator 540
  - spark-k8s-operator 513
  - superset-operator 584
  - trino-operator 691
  - zookeeper-operator 903

File: docs/templating_vars.yaml [hunk 1]
```diff
@@ -3,9 +3,9 @@ helm:
...
   repo_url: https://repo.stackable.tech/repository/helm-stable/
 versions:
-  commons: 24.11.1-rc2
-  secret: 24.11.1-rc2
-  listener: 24.11.1-rc2
-  airflow: 24.11.1-rc2
+  commons: 24.11.1
+  secret: 24.11.1
+  listener: 24.11.1
+  airflow: 24.11.1
```
Comment (NickLarsenNZ) [inline docs/templating_vars.yaml:9]: Note to self, we should check the regexes across the scripts to ensure they account for pre-release info. Maybe that's already been done.

**Review by NickLarsenNZ — APPROVED** [review]
LGTM

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -350,13 +350,15 @@ jobs:
...
+        # We can't rely on cargo set-version here as we will break semver rules when changing the version to make it
+        # specific to this PR e.g. 1.2.0 --> 1.2.0-pr678, so set it manually.
         if: ${{ github.event_name == 'pull_request' && github.ref != 'refs/heads/main' }}
         env:
           PR_NUMBER: ${{ github.event.pull_request.number }}
         run: |
           MANIFEST_VERSION=$(cargo metadata --format-version 1 --no-deps | jq -r '.packages[0].version')
           PR_VERSION="${MANIFEST_VERSION}-pr${PR_NUMBER}"
-          cargo set-version --offline --workspace "$PR_VERSION"
+          sed -i "s/version = \"${MANIFEST_VERSION}\"/version = \"${PR_VERSION}\"/" Cargo.toml
```
[the remark this answers is not in the document]
Comment (adwk67) [reply .github/workflows/build.yml:361]: See https://github.com/stackabletech/operator-templating/pull/475

File: docs/templating_vars.yaml (hunk 1 above)
Comment (adwk67) [reply docs/templating_vars.yaml:9]: I don't think the getting-started tests will work prior to release-tagging, as the images will only exist in their `-prXXX` state.

Comment (adwk67) [reply docs/templating_vars.yaml:9]: We would have to manually edit that file to test that, much like we have done for the integration tests (see the release.yaml changes etc.)

Comment (NickLarsenNZ) [reply docs/templating_vars.yaml:9]: Will mark this as resolved, as I do think the regexes cover the pre-release info.
